apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,180 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// BackendPolicy defines policies associated with backends. For the purpose of
+// this API, a backend is defined as any resource that a route can forward
+// traffic to. A common example of a backend is a Service. Configuration that is
+// implementation specific may be represented with similar implementation
+// specific custom resources.
+type BackendPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BackendPolicySpec   `json:"spec,omitempty"`
+	Status BackendPolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// BackendPolicyList contains a list of BackendPolicy
+type BackendPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendPolicy `json:"items"`
+}
+
+// BackendPolicySpec defines desired policy for a backend.
+type BackendPolicySpec struct {
+	// BackendRefs define which backends this policy should be applied to. This
+	// policy can only apply to backends within the same namespace. If more than
+	// one BackendPolicy targets the same backend, precedence must be given to
+	// the oldest BackendPolicy.
+	//
+	// Support: Core
+	// +kubebuilder:validation:MaxItems=16
+	BackendRefs []BackendRef `json:"backendRefs"`
+
+	// TLS is the TLS configuration for these backends.
+	//
+	// Support: Extended
+	TLS *BackendTLSConfig `json:"tls"`
```

Comment:
Why is this a pointer if this is not optional?
Should this be optional? I think we discussed this before and decided against it because it is the only policy we have currently. Does this mean we will make it optional in future if we have something like health-checking in here?

Reply:
Good point, this really should be optional, updated.

Reply:
This needs `omitempty` now.

Reply:
Updated, thanks!